File: ClientApp/src/app/clases/docentemateria.ts
export class docentemateria{
    id : number;
     
    iddocente : number;
    idmateria : number;
    idciclolectivo : number;
    
    constructor(obj:any){
        //el orden de os parametros afecta la vista de los input en el modal
        this.id = +obj.id;
        this.iddocente = +obj.iddocente;
        this.idmateria =  +obj.idmateria;
        this.idciclolectivo = +obj.idciclolectivo;
       
         // formateo a number el string // parseInt(val

    }
    
}

File: modelo/docentemateria.cs
using System.Data;
using System;
public class docentemateria : oObjeto
{
    public int Iddocente { get; set; }
    public int Idmateria { get; set; }
    public int Idciclolectivo { get; set; }
    public docentemateria() { }
    public docentemateria(DataRow dr)
    { 
        this.Iddocente = Convert.ToInt32(dr["iddocente"]);
        this.Idmateria = Convert.ToInt32(dr["idmateria"]);
        this.Idciclolectivo = Convert.ToInt32(dr["idciclolectivo"]);
    }

}